lib/resources/postgres_session.rb

```diff
@@ -39,25 +41,32 @@ class PostgresSession < Inspec.resource(1)
         its('output') { should eq('') }
       end
     "
+    attr_reader :user, :pass, :host, :db, :escaped_query, :psql_cmd
 
     def initialize(user, pass, host = nil)
       @user = user || 'postgres'
       @pass = pass
       @host = host || 'localhost'
     end
 
-    def query(query, db = [])
+    def escaped_query(query)
+      return warn 'no query specificed' if query.empty? || query.nil?
+      Shellwords.escape(query)
+    end
+
+    def create_psql_cmd(query, db = [])
+      @db = db
       dbs = db.map { |x| "-d #{x}" }.join(' ')
-      # TODO: simple escape, must be handled by a library
-      # that does this securely
-      escaped_query = query.gsub(/\\/, '\\\\').gsub(/"/, '\\"').gsub(/\$/, '\\$')
-      # run the query
-      cmd = inspec.command("PGPASSWORD='#{@pass}' psql -U #{@user} #{dbs} -h #{@host} -A -t -c \"#{escaped_query}\"")
+      @escaped_query = escaped_query(query)
+      "PGPASSWORD='#{@pass}' psql -U #{@user} #{dbs} -h #{@host} -A -t -c #{@escaped_query}"
+    end
+
+    def query(query, db = [])
+      @db = db
```

Comment:
I mainly wanted to be able to test the main parts of the construction process to what I was passing to psql. i.e. that the query escaped correctly with the corner case tests - like : SELECT current_setting('client_min_messages') . 

I guess if I just expose @psql_cmd I would be able to look at the parts of that and ensure that the functions are all producing what we are expecting.